### ISSUE TITLE
Issue with Interpreting Codeml Output from PoSeiDON

### ISSUE THREAD

**Issue by Dx-wmc** [rnajena/poseidon#37]
I am using PoSeiDON for positive selection analysis, as mentioned in a recent publication (https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1012161). The software successfully generated multiple codeml output files (.mlc files), but I am having trouble interpreting them and applying the empirical Bayes approach mentioned in the paper: "we used an empirical Bayes approach to calculate posterior probabilities that a codon coming from a site class with dN/dS>1. Genes were considered to be positively selected when the p-value is <0.05 and the dN/dS ratio exceeds one."

Could you provide guidance or examples on how to proceed with analyzing these .mlc files to identify positively selected genes based on this approach?

**Comment by hoelzer** [comment]
Hey @Dx-wmc thanks for using the pipeline!

Yes, the `*.mlc` files are tricky. That's why we are also producing these summary `*.pdf` files. 

Basically, the BEB (Bayes Empircal Bayes) calulations and the posterior probabilities (PP) are done by `CODEML` and within the pipeline. Here is the outptu from the example data summarized in the `bats_mx1_small_codeml.pdf` file:

![grafik](https://github.com/user-attachments/assets/27c47bf4-ecd2-488d-ade2-abd840f868ed)

See also the caption. 

This table summarizes all tests done within CODEML using different models. And you can also see which sites (codons) having a BEB PP larger 0.95 or 0.99 overlap between the different model tests. 

So in short: probably you dont have to analyse the `.mlc` files yourself bc the pipeline does that for you, extracts the interesting parameters, and summarizes them. 

I hope that helps!

**Comment by Dx-wmc** [comment]
Thank you for your  patient answer. However, my question here focuses on the entire gene rather than the site, so I may still need to further analyze the mlc file.

**Comment by Dx-wmc** [comment]
Thank you! Your explanation has enlightened me. It seems that I have been going in the wrong direction before.
